ngc-monkey/ngc-monkey/traffic_generator.py

```diff
@@ -0,0 +1,68 @@
+import time
+import typing
+from enum import Enum
+
+from threading import Thread, Event
+import logging
+
+from .raft_client import RaftClient
+module_logger = logging.getLogger(__name__)
+
+
+class Result(str, Enum):
+    SUCCESS = "success"
+    FAILURE = "failure"
+
+    def is_success(self) -> bool:
+        return self == self.SUCCESS
+
+
+class TrafficGenerator(Thread):
+
+    def __init__(self, host: str):
+        super(TrafficGenerator, self).__init__()
+        self._event = Event()
+        self.host = host
+        self.redis_con = RaftClient(host=host, decode_responses=True)
+        self.redis_con.ping()
+        self._memory: typing.Dict[str, str] = {}
+        self.result: typing.Optional[Result] = None
+        self.logger = None
+
+    def stop_and_check(self):
+        """
+        Tell the traffic generator to stop sending traffic and check data integrity
+        """
+        self._event.set()
+
+    def _check(self) -> Result:
```

Comment:
is there an issue with threads visibility or atomicity here? when u set the event from one thread ( the http thread ) and u check in the generator thread?

Reply:
fixing